Page::m_mainFrame should be an AbstractFrame

achristensen07 opened the pull request:
#### dfb90751211591c4b39efb57ebd798186f7a2338
<pre>
Page::m_mainFrame should be an AbstractFrame
<a href="https://bugs.webkit.org/show_bug.cgi?id=252840">https://bugs.webkit.org/show_bug.cgi?id=252840</a>
rdar://105839964

Reviewed by Chris Dumez.

* Source/WebCore/page/Page.cpp:
(WebCore::Page::~Page):
(WebCore::Page::scrollingStateTreeAsText):
(WebCore::Page::synchronousScrollingReasonsAsText):
(WebCore::Page::nonFastScrollableRectsForTesting):
(WebCore::Page::touchEventRectsForEventForTesting):
(WebCore::Page::passiveTouchEventListenerRectsForTesting):
(WebCore::Page::goToItem):
(WebCore::Page::analyzeImagesForFindInPage):
(WebCore::Page::layoutIfNeeded):
(WebCore::Page::mediaSessionGroupIdentifier const):
(WebCore::Page::setActivityState):
(WebCore::Page::setSessionID):
(WebCore::Page::dispatchBeforePrintEvent):
(WebCore::Page::dispatchAfterPrintEvent):
(WebCore::Page::configureLoggingChannel):
(WebCore::Page::injectUserStyleSheet):
(WebCore::Page::removeInjectedUserStyleSheet):
(WebCore::Page::mainFrameDidChangeToNonInitialEmptyDocument):
(WebCore::Page::reloadExecutionContextsForOrigin const):
* Source/WebCore/page/Page.h:

Canonical link: <a href="https://commits.webkit.org/260767@main">https://commits.webkit.org/260767@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cd72803229b909024103442410b10b214a8ff1f8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [  ~~🧪 style~~](https://ews-build.webkit.org/#/builders/6/builds/109292 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/77/builds/18372 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/42089 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/87/builds/806 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/118510 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/11/builds/113174 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/76/builds/19821 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/85/builds/9662 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/101532 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/19/builds/115047 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/76/builds/19821 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/42089 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/101532 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/76/builds/19821 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/42089 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/101532 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/11146 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/42089 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/11864 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/85/builds/9662 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/17257 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/42089 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/79/builds/13497 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/page/Page.cpp:
```diff
@@ -405,7 +405,8 @@ Page::~Page()
     m_performanceLoggingClient = nullptr;
-    m_mainFrame->setView(nullptr);
+    if (auto* localMainFrame = dynamicDowncast<LocalFrame>(m_mainFrame.get()))
+        localMainFrame->setView(nullptr);
```
Didn't the RemoteFrame have a RemoteView? Maybe I remember wrong.

cdumez reviewed (approved):
LGTM

achristensen07 committed:
Page::m_mainFrame should be an AbstractFrame
https://bugs.webkit.org/show_bug.cgi?id=252840
rdar://105839964

Reviewed by Chris Dumez.

* Source/WebCore/page/Page.cpp:
(WebCore::Page::~Page):
(WebCore::Page::scrollingStateTreeAsText):
(WebCore::Page::synchronousScrollingReasonsAsText):
(WebCore::Page::nonFastScrollableRectsForTesting):
(WebCore::Page::touchEventRectsForEventForTesting):
(WebCore::Page::passiveTouchEventListenerRectsForTesting):
(WebCore::Page::goToItem):
(WebCore::Page::analyzeImagesForFindInPage):
(WebCore::Page::layoutIfNeeded):
(WebCore::Page::mediaSessionGroupIdentifier const):
(WebCore::Page::setActivityState):
(WebCore::Page::setSessionID):
(WebCore::Page::dispatchBeforePrintEvent):
(WebCore::Page::dispatchAfterPrintEvent):
(WebCore::Page::configureLoggingChannel):
(WebCore::Page::injectUserStyleSheet):
(WebCore::Page::removeInjectedUserStyleSheet):
(WebCore::Page::mainFrameDidChangeToNonInitialEmptyDocument):
(WebCore::Page::reloadExecutionContextsForOrigin const):
* Source/WebCore/page/Page.h:

Canonical link: https://commits.webkit.org/260767@main
